README.md

```diff
@@ -1 +1,77 @@
+
+Supported operating systems:
+
+  * `macos`
+
+Supported architectures:
+
+  * `amd64`
+  * `arm64`
+
+The goal of these builds is to be as self-contained as possible. OpenSSL (used
+by `crypto` app) and wxWidgets (used by `wx` app and it's dependants,
+`observer`, `debugger`, and `et`) are statically linked.
+
+## List of Builds
+
+| Target                      | OpenSSL | wxWidgets                 |
+| --------------------------- | ------- | ------------------------- |
+| **OTP 25**                  |         |                           |
+| `macos-amd64`               | 3.1.6   | N/A <sup>2</sup>          |
+| `macos-arm64` <sup>1</sup>  | 3.1.6   | N/A <sup>2</sup>          |
+| **OTP 26**                  |         |                           |
+| `macos-amd64`               | 3.1.6   | N/A <sup>2</sup> / 3.2.5  |
+| `macos-arm64`               | 3.1.6   | N/A <sup>2</sup> / 3.2.5  |
+| **OTP 27**                  |         |                           |
+| `macos-amd64`               | 3.1.6   | 3.2.5                     |
+| `macos-arm64`               | 3.1.6   | 3.2.5                     |
+| **OTP maint** <sup>3</sup>  |         |                           |
+| `macos-amd64`               | 3.1.6   | 3.2.5                     |
+| `macos-arm64`               | 3.1.6   | 3.2.5                     |
+| **OTP master** <sup>3</sup> |         |                           |
+| `macos-amd64`               | 3.1.6   | 3.2.5                     |
+| `macos-arm64`               | 3.1.6   | 3.2.5                     |
+
+<sup>1</sup> JIT is disabled on OTP 25 on macos-arm64.
+
+<sup>2</sup> `wx`, `observer`, `debugger`, and `et` apps are only available since OTP 26.1.1.
+
+<sup>3</sup> OTP maint and master builds are updated daily.
+
+List of builds can be found here:
+
+  * `macos-amd64` - <https://github.com/erlef/otp_builds/raw/builds/macos-amd64.txt>
+  * `macos-arm64` - <https://github.com/erlef/otp_builds/raw/builds/macos-arm64.txt>
+
+Entries in the list of builds follow this pattern:
+
+    {ref_name} {ref} {datetime} {sha256}
+
+Where `{ref_name}` is the Erlang/OTP git tag or branch name, `{ref}` is the git sha corresponding
+to the `{ref_name}`, `{datetime}` is the time the build was created and `{sha256}` is the build
+SHA-256 checksum.
+
+Example `builds/macos-arm64.txt` entry:
+
+    OTP-27.0.1 ee9628e7ed09ef02e767994a6da5b7a225316aaa 2024-09-12T10:53:59Z a805cd23cd5362a294d1e88fa0919192b0f4cbe5601f906167c11a2b57c98ac4
+
+## Build Downloads
+
+Build download URLs follow this pattern:
+
+    https://github.com/erlef/otp_builds/releases/download/{ref_name}/{ref_name}-macos-{arch}.tar.gz
+
+Where `{ref_name}` is the name of Erlang/OTP git tag or branch name and `{arch}` is the architecture.
+The supported Erlang/OTP branches are `master` and `maint`.
+
+Example build URLs:
+
+  * <https://github.com/erlef/otp_builds/releases/download/OTP-27.0.1/OTP-27.0.1-macos-arm64.tar.gz>
```

Comment:
> `OTP-27.0.1-macos-arm64.tar.gz`

I'd like to get some feedback on the naming scheme.

Ideally we'd use naming that'd be be easy to arrive at using basic tooling, like `uname`, however speaking of that it is so inconsistent:

- macOS arm64: `bash$ uname -sm #=> Darwin arm64`
- Linux arm64: `bash$ uname -sm #=> Linux aarch64`
- Windows x86_64: `bash$ uname -sm #=> MINGW64_NT-10.0 x86_64`, `powershell> echo $env:PROCESSOR_ARCHITECTURE #=> AMD64`

so I think nevermind that.

I have found these naming schemes:

1. target triple, e.g. from `rustc --print target-list`

    - `aarch64-apple-darwin`
    - `aarch64-apple-ios`
    - `aarch64-unknown-linux-gnu`
    - `aarch64-unknown-freebsd`

   (No plans for iOS or FreeBSD builds here _at the moment_.)

2. `$GOOS`/`$GOARCH`, `go tool dist list`:

    - `darwin-arm64`
    - `linux-arm64`
    - `freebsd-arm64`

As an aside, existing hexpm/bob builds, https://github.com/hexpm/bob?tab=readme-ov-file#erlang-builds use amd64/arm64 architecture identifier. I hope to eventually move them to this repor. Unless we could have Linux builds that are distribution independent (perhaps just libc dependent), we'd have to put the Ubuntu name in there _anyway_, say `OTP-27.0.1-linux-arm64-ubuntu-22.04.tar.gz` so it seems like a custom naming scheme too.

As an another aside, official OTP Windows builds are named e.g. `otp_win64_27.1.zip`.

Finally, there's a matter of `:erlang.system_info(:system_architecture)`:

  - macOS arm64 (built-on macOS 14):

        elixir -e "IO.puts :erlang.system_info(:system_architecture)"
        aarch64-apple-darwin23.5.0

  - macOS x64 (built-on macOS _13_):

        elixir -e "IO.puts :erlang.system_info(:system_architecture)"
        x86_64-apple-darwin22.6.0

  - Linux arm64:

        $ docker run --rm -it hexpm/elixir:1.17.3-erlang-27.1.2-ubuntu-jammy-20240808 elixir -e "IO.puts :erlang.system_info(:system_architecture)"
        aarch64-unknown-linux-gnu

  - Linux x64:

        $ docker run --rm -it hexpm/elixir:1.17.3-erlang-27.1.2-ubuntu-jammy-20240808 elixir -e "IO.puts :erlang.system_info(:system_architecture)"
        x86_64-pc-linux-gnu

  - Windows:

        $ elixir -e "IO.puts :erlang.system_info(:system_architecture)"
        win32


Worth reminding this is the target triple (not a triple on Windows) of the _build_, not of the currently running OS. I'm not sure if `23.5.0` in `aarch64-apple-darwin23.5.0` is useful btw. I'm making these builds with `-mmacosx-version-min=11.0`, so if anything, following `swiftc -print-target-info | jq .target.triple`, it could be `arm64-apple-macosx11.0`. `swiftc a.swift -target aarch64-apple-macos11.0` also works. I think `macos` in the target triple is nicer but `darwin` as a name is probably still more pervasive.

Not sure if we need to address both build file naming and OTP `system_architecture` but I think for these builds we could do the following which shouldn't hurt:

- `OTP-27.1.2-aarch64-apple-darwin.tar.gz` `system_architecture` reports `aarch64-apple-darwin`
- `OTP-27.1.2-x86_64-apple-darwin.tar.gz` `system_architecture` reports `x86_64-apple-darwin`

When we add Ubuntu we could do this:

- `OTP-27.1.2-aarch64-pc-linux-ubuntu-24.04.tar.gz` `system_architecture` reports `aarch64-pc-linux` (or `unknown-linux`?)
- `OTP-27.1.2-x86_64-pc-linux-ubuntu-22.04.tar.gz` `system_architecture` reports `x86_64-pc-linux`

or, ideally, Linux distribution-independent builds:

- `OTP-27.1.2-aarch64-pc-linux.tar.gz` `system_architecture` reports `aarch64-pc-linux`
- `OTP-27.1.2-x86_64-pc-linux.tar.gz` `system_architecture` reports `x86_64-pc-linux`

WDYT?